src/ui/AnotherQuickSwitcherModal.ts

```diff
@@ -90,6 +90,41 @@ export class AnotherQuickSwitcherModal
     this.scope.register(["Mod"], "K", () => {
       document.dispatchEvent(new KeyboardEvent("keydown", { key: "ArrowUp" }));
     });
+    // Delete whole input line
+    this.scope.register(["Mod"], "D", () => {
```

Comment:
It worked by dispatching the "input" event.

```
this.scope.register(["Mod"], "D", () => {
      this.inputEl.select();
      this.inputEl.setRangeText("");
      this.inputEl.dispatchEvent(new Event("input"));
});
```

However, it depends on the implementation of Obsidian, so there is a possibility not to work someday.

Reply:
It worked by dispatching the "input" event.

```typescript
this.scope.register(["Mod"], "D", () => {
      this.inputEl.select();
      this.inputEl.setRangeText("");
      this.inputEl.dispatchEvent(new Event("input"));
});
```

>However, it depends on the implementation of Obsidian, so there is a possibility not to work someday.

That is a nice trick! It works fine, thank you. I have updated the PR according to your suggestions.